db/database.go

```diff
@@ -152,8 +155,12 @@ func ConnectToBucket(spec base.BucketSpec, callback func(bucket string, err erro
 	return
 }
 
+// Function type for something that calls NewDatabaseContext and wants a callback when the DB is detected
+// to come back online. A rest.ServerContext package cannot be passed since it would introduce a circular dependency
+type DBOnlineCallback func(dbContext *DatabaseContext)
+
 // Creates a new DatabaseContext on a bucket. The bucket will be closed when this context closes.
-func NewDatabaseContext(dbName string, bucket base.Bucket, autoImport bool, options DatabaseContextOptions) (*DatabaseContext, error) {
+func NewDatabaseContext(dbName string, bucket base.Bucket, autoImport bool, options DatabaseContextOptions, online DBOnlineCallback) (*DatabaseContext, error) {
```

Comment:
I've been trying to push modifications to the NewDatabaseContext API into DatabaseContextOptions, to avoid bloat on the signature (and having all the test callers pass a nil value that's not self-descriptive).  Do you think the callback could be passed via the options as well?